### PR TITLE
Add explicit dependency info to enable FastUpToDateCheck feature in VS

### LINKED ISSUE #804: C#/WinRT Interop projects are always up-to-date regardless of their project dependencies
**Summary**:
C#/WinRT Interop projects are considered up-to-date after changing C++/WinRT Sources when fast Up To Date checking is enabled

**Steps to Reproduce**:
1. Launch Visual Studio 2019 and go to Tools > Options > Projects and Solutions > SDK-Style Projects
2. Ensure this option is checked: "Don't call MSBuild if a project appears to be up to date."
3. Go to Extensions > Manage Extensions
4. Search for “C++” then select C++/WinRT and click on “Download”
5. Restart Visual Studio to complete the installation
6. Create a new Windows Runtime Component (C++/WinRT) (called “CPPWinRTInteropRepro”, for example)
7. Click OK to accept the default target and minimum platform versions (e.g. 19041 and 17763)
8. Add a new C# Class Library (.NET Core) project called “CPPWinRTInteropReproProjection” to your solution.
9. Delete the empty Class1.cs file from the project.
10. Enable “Show All Files” for both projects in the solution explorer toolbar.
11. In Solution Explorer, right click your class library project and select “Manage NuGet Packages”.
12. Search for the “Microsoft.Windows.CsWinRT” NuGet package and install the latest version -  https://www.nuget.org/packages/Microsoft.Windows.CsWinRT
13. In Solution Explorer, right click the class library project and select “Edit Project File”.
14. Update the TargetFramework element to reference the Windows SDK, e.g. “net5.0-windows10.0.19041.0”
15. Add a new PropertyGroup to the project file with 
  `<PropertyGroup>
    <CsWinRTIncludes>CPPWinRTInteropRepro</CsWinRTIncludes>
    <CsWinRTWindowsMetadata>10.0.19041.0</CsWinRTWindowsMetadata>
  </PropertyGroup>`
16. Save and close the CPPWinRTInteropReproProjection.csproj file.
17. In Solution Explorer, right click the “Dependencies” node under the CPPWinRTInteropReproProjection project, select “Add Project Reference”, and select the CPPWinRTInteropRepro project.
18. Add a new C# .NET 5 console application (e.g. ConsoleApp1).
19. In Solution Explorer, right click the console application project and select “Edit Project File”.
20. Change the TargetFramework to “net5.0-windows10.0.19041.0” to match the projection project.
21. Add this property to ConsoleApp1: `<PropertyGroup><CsWinRTWindowsMetadata>10.0.19041.0</CsWinRTWindowsMetadata></PropertyGroup>`
22. Save and close the ConsoleApp1.csproj editor/window.
23. Install the “Microsoft.Windows.CsWinRT” NuGet package into ConsoleApp1.
24. In Solution Explorer, right click the “Dependencies” node under the ConsoleApp1project, select “Add Project Reference”, and select the CPPWinRTInteropReproProjection project.
25. Add this code to Program.cs: `var someObject = new Class();`
26. Add a using statement to the top of the file: `using CPPWinRTInteropRepro;`
27. Build ConsoleApp1. This should succeed.
28. Open Class.idl and add a new property (i.e Int32 NewProperty)
29. Build the C++ project.
30. Copy the generated class.h and class.cpp files into the main CPP project (copy the files from the folder Generated Files\sources to CPPWinRTInteropRepro ? and replace them ?).
31. Remove the static asserts that cause build to fail (comment the line that contains static_assert in class.cpp and class.h)
32. Build the C++ project, this should succeed.
33.   Build the C++ again, until the project is marked up-to-date
34. Reference the newly added property in ConsoleApp1 (Add the statement someObject.NewProperty = 0 to Program.cs).
35. Build ConsoleApp1.

**Expected Behavior**:
Building the console app (which depends on the interop project) should succeed.

**Actual Behavior**:
Building the console app fails.

**User Impact**:
Users need to rebuild the interops apps manually after changing the C++/WinRT sources.

**C++/WinRT Resources**:
1. https://docs.microsoft.com/en-us/windows/uwp/cpp-and-winrt-apis/intro-to-using-cpp-with-winrt
2. https://docs.microsoft.com/en-us/windows/uwp/csharp-winrt/net-projection-from-cppwinrt-component

**Version Info**
- CsWinRT 1.2.1
- Visual Studio Version**: 16.9.0

**Additional context**
Copied from https://github.com/dotnet/project-system/issues/7067

### PR DESCRIPTION
**Pull request by Scottj1s**
fixes #804

### CODE REVIEW

**Comment by Scottj1s** [comment]
FYI:
https://github.com/dotnet/project-system/blob/main/docs/up-to-date-check.md

File: nuget/Microsoft.Windows.CsWinRT.targets [hunk 1]
```diff
@@ -107,6 +107,13 @@ $(CsWinRTFilters)
     <WriteLinesToFile File="$(CsWinRTResponseFile)" Lines="$(CsWinRTParams)" Overwrite="true" WriteOnlyWhenDifferent="true" />
     <Message Text="$(CsWinRTCommand)" Importance="$(CsWinRTMessageImportance)" />
     <Exec Command="$(CsWinRTCommand)" />
+    <ItemGroup>
+      <CsWinRTOutputs Include="$(CsWinRTGeneratedFilesDir)/*.cs"/>
+      <UpToDateCheckInput Include="@(CsWinRTInputs)" Set="CsWinRTInputs" />
```
Comment (manodasanW) [inline nuget/Microsoft.Windows.CsWinRT.targets:112]: In the doc I see `Note that UpToDateCheckOutput exists but is deprecated and only maintained for backwards compatability. Projects should use to UpToDateCheckBuilt instead.`  Thoughts?

Comment (Scottj1s) [reply nuget/Microsoft.Windows.CsWinRT.targets:112]: yes, I saw that too. but the UpToDateCheckBuilt doesn't support grouped dependencies (and even goes on to say that it doesn't compose with sets, implying that UpToDateCheckOuput *is* still supported)

Comment (Scottj1s) [reply nuget/Microsoft.Windows.CsWinRT.targets:112]: the worse case is that we're no better off than before, if VS ever decides to stop supporting (i.e, there's no risk of making things worse)

**Review by manodasanW — APPROVED** [review]

**Pull request merged**